src/pool/srv_iv.c

```diff
@@ -434,6 +439,8 @@ pool_iv_ent_refresh(struct ds_iv_entry *entry, struct ds_iv_key *key,
 
 	if (entry->iv_class->iv_class_id == IV_POOL_PROP)
 		rc = ds_pool_tgt_prop_update(pool, &src_iv->piv_prop);
+	else if (entry->iv_class->iv_class_id == IV_POOL_CONN)
+		rc = ds_pool_tgt_connect(pool, &src_iv->piv_conn);
```

Comment:
I can understand distributing data like pool map through IV, but I really don't see the value of making these stateful sync operations like open/close/connect/disconnect as IV. (We've already hit some bugs caused by binding container open with IV refresh). Is there anything important I missed?

Is it for distributing operations (open handles) to later joined/added node through IV? I personally think this idea will make IV too complicated. In my opinion, that should be done by explicit operation through a special reintegration procedure, and operation race or failure needs be handled by reintegration. Keeping IV away from the reintegration logic could make code easier to maintain, what do you think?

Reply:
Personally, I do not think IV complicate any logic, though the implementation might not be simple and clear enough for the moment.

For me IV is more efficient and flexible for controlling the flow of information,  especially for adding new node.

The reintegration will be explicit anyway, I do not think IV will bury this process.  Or could you please be more specific about the concern?

Reply:
We do plan to move pool handles (i.e., connections) to IV. It could make the pool service operations more "local"---just update the pool connection IV root to add a new connection and let IV disseminate the update asynchronously. That said, I share the concern on potential regressions, considering the move of container handles to IV.

Reply:
I think there are two kinds of pool/container connect/open protocol (I mean server target connect/open):
1. All targets do explicit synchronous connect/open when client does pool/container connect/open;
2. Each target does on-demand connect/open independently, no explicit target disconnect/close is required;

I think IV is only valuable for the 2nd protocol, and we are using 1st protocol now. Are we going to change to use 2nd protocol? Sorry, I just want to fully understand the motivation of this kind of change.

Reply:
Yes, we will go 2nd path. those changes will be added in the following patches.